x/qgb/abci_test.go

```diff
@@ -111,13 +111,88 @@ func TestSetDataCommitment(t *testing.T) {
-	vs, err := qk.GetCurrentDataCommitment(ctx)
+	ctx = ctx.WithBlockHeight(int64(qk.GetDataCommitmentWindowParam(ctx)))
+	dc, err := qk.GetCurrentDataCommitment(ctx)
+	require.NoError(t, err)
+	err = qk.SetAttestationRequest(ctx, &dc)
+	require.NoError(t, err)
+
+	require.Equal(t, uint64(1), qk.GetLatestAttestationNonce(input.Context))
+}
+
+// TestGetDataCommitment This test will test the create of data commitment ranges
+// in the event of the data commitment window changing via an upgrade or a gov proposal.
+// The test goes as follows:
+//   - Start with a data commitment window of 400
+//   - Get the first data commitment, its range should be: [1, 400]
+//   - Get the second data commitment, its range should be: [401, 800]
+//   - Shrink the data commitment window to 101
+//   - Get the third data commitment, its range should be: [801, 901]
+//   - Expand the data commitment window to 500
+//   - Get the fourth data commitment, its range should be: [902, 1401]
+func TestGetDataCommitment(t *testing.T) {
+	input, ctx := testutil.SetupFiveValChain(t)
+	qk := input.QgbKeeper
+
+	dcWindow := qk.GetDataCommitmentWindowParam(ctx)
+
+	// get the first data commitment
+	ctx = ctx.WithBlockHeight(int64(dcWindow))
+	dc1, err := qk.GetCurrentDataCommitment(ctx)
+	require.NoError(t, err)
+	require.Equal(t, uint64(1), dc1.BeginBlock)
+	require.Equal(t, uint64(dcWindow), dc1.EndBlock)
+	require.Equal(t, uint64(1), dc1.Nonce)
+	// set the first data commitment
+	err = qk.SetAttestationRequest(ctx, &dc1)
+	require.Nil(t, err)
+
+	// get the second data commitment
+	ctx = ctx.WithBlockHeight(int64(dcWindow) * 2)
+	dc2, err := qk.GetCurrentDataCommitment(ctx)
 	require.Nil(t, err)
-	err = qk.SetAttestationRequest(ctx, &vs)
+	require.Equal(t, dc1.EndBlock+1, dc2.BeginBlock)
+	require.Equal(t, dc1.EndBlock+dcWindow, dc2.EndBlock)
+	require.Equal(t, uint64(2), dc2.Nonce)
+	// set the second data commitment
+	err = qk.SetAttestationRequest(ctx, &dc2)
 	require.Nil(t, err)
 
-	require.Equal(t, uint64(1), qk.GetLatestAttestationNonce(ctx))
+	// shrink the data commitment window
+	genesis := types.DefaultGenesis()
+	newShrinkedDCWindow := dcWindow - 299 // 101, since the default one is 400
+	genesis.Params.DataCommitmentWindow = newShrinkedDCWindow
+	qk.SetParams(ctx, *genesis.Params)
+	require.Equal(t, newShrinkedDCWindow, qk.GetDataCommitmentWindowParam(ctx))
+
+	// get the third data commitment window
+	wantedHeight := nextMultiple(int64(newShrinkedDCWindow), int64(dcWindow)*2)
+	// to simulate the condition in endBlocker: ctx.BlockHeight()%int64(k.GetDataCommitmentWindowParam(ctx))
+	ctx = ctx.WithBlockHeight(wantedHeight)
+	dc3, err := qk.GetCurrentDataCommitment(ctx)
+	require.Nil(t, err)
+	require.Equal(t, dc2.EndBlock+1, dc3.BeginBlock)
+	require.Equal(t, dc2.EndBlock+newShrinkedDCWindow, dc3.EndBlock)
+	require.Equal(t, uint64(3), dc3.Nonce)
+	// set the third data commitment
+	err = qk.SetAttestationRequest(ctx, &dc3)
```

Comment:
as mentioned in a different reviewer's comment, parsing this is difficult because its difficult to understand what criteria is being checked. so we could perhaps add a few comments to elaborate on what we are looking for and why.

Reply:
What do you think about this explanation? [90593fd](https://github.com/celestiaorg/celestia-app/pull/1695/commits/90593fd72a94445371786f6e49970704996e4d7c).
If not clear enough, I will inline comments with every line of the test. 
Let me know what you think is better.